.github/workflows/frontend-build-lint-typecheck.yml

```diff
@@ -0,0 +1,33 @@
+name: Frontend build, lint, and typecheck
+
+on:
+  push:
+    paths-ignore:
+      - "backend/**"
+
+jobs:
+  build-lint-typecheck:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+
+      - name: Enable Corepack
+        run: corepack enable
+
+      - name: Install Dependencies
+        run: yarn install
+
+      - name: Build
+        run: yarn build
+        
+      - name: Lint
+        run: yarn lint
+        
+      - name: Typecheck
+        run: yarn typecheck
```

Comment:
Might be a preference-thing, but I really like separating the lint and typechecking from the build to make it easier to see in the landing-page on the pull-request what part of the workflow actually failed. That means you have to checkout the code multiple times tho, so I guess there are pros and cons for both approaches.